### ISSUE TITLE
Is boundary_azi deprecated or is AziBoundaryLoss class only partially implemented?

### ISSUE THREAD

**Issue by ackl** [RobertJaro/NF2#31]
firstly, thanks for building NF2 it's really really cool!

i was checking the azimuth boundary losses and seems like `AziBoundaryLoss` may be borked for the 180 deg ambiguity

`AziBoundaryLoss` currently does:

```python
b_azi_true = b_true[..., 2] % torch.pi
b_azi_pred = b_pred[..., 2] % torch.pi
b_diff = (b_azi_pred - b_azi_true).pow(2) * b_true[..., 1]
```
@ azimuths 0.01 and pi - 0.01 `b_diff` should be close under the 180 degree ambiguity, but this loss treats them as almost pi apart:
```
import torch

a = torch.tensor(0.01)
b = torch.tensor(torch.pi - 0.01)

current = ((a % torch.pi) - (b % torch.pi)).pow(2)
expected = torch.sin(a - b).pow(2)

print(current)   # ~9.74
print(expected)  # ~0.0004
```

so near the 0/pi wrap boundary the optimiser gets a large gradient in the wrong direction, which i think can inject artificial azimuth discontinuities into extrapolations if a run is configured with a loss with type `boundary_azi`

constructor method for `AziBoundaryLoss` accepts disambiguate=True and stores it into `self.disambiguate` but the value is never used

By contrast `LosTrvAziBoundaryLoss(ambiguous=True)` does handle the ambiguity in Cartesian transverse components using `abs(dot)` which avoids this wraparound issue

Is `boundary_azi` intended to be deprecated / legacy, with `boundary_los_trv_azi` as the supported path? cos i don't see `boundary_azi` in any of the prewritten yml configs in the repo, and [the auto_disambiguation.yaml](https://github.com/RobertJaro/NF2/blob/main/nf2/configs/cartesian/auto_disambiguation.yaml#L48) example config uses `boundary_los_trv_azi`

or was`AziBoundaryLoss` intended to support disambiguation directly and the implementation was just unfinished and missed? if so i'd be happy to open a PR

**Comment by RobertJaro** [comment]
Thanks for catching this. Yes, boundary_azi is mostly legacy. The currently path for ambiguity-aware LOS/transverse/azimuth constraints is boundary_los_trv_azi, as used in the bundled configs.

That said, AziBoundaryLoss does expose disambiguate=True and the modulo implementation is wrong, so the previous behavior was confusing and incorrect. I updated this so that the disambiguated case now uses a pi-periodic azimuth loss, which makes azimuths close across the 0/pi wrap boundary produce a small loss under the 180-degree ambiguity. The loss is still scaled by the transverse field strength.
```python
delta = b_azi_pred - b_azi_true
loss = torch.sin(delta).pow(2) * b_true[..., 1]
```
or the non-disambiguated case, the loss uses the shortest full-angle circular difference, so it remains 2*pi periodic without the artificial wrap jump:
```python
delta = b_azi_pred - b_azi_true
b_delta = torch.atan2(torch.sin(delta), torch.cos(delta))
loss = b_delta.pow(2) * b_true[..., 1]
```
Please have a look if this implementation works for you. If you have a specific implementation or use case in mind, feel free to submit a PR.
